Minimal pair task, maskLM acceptability, relocate NPI task

HaokunLiu opened the pull request:
Introducing minimal pair task, its input is a pair of sentences, but it does not inherit from pair classification task, otherwise bert model will use "[CLS] sent1 [SEP] sent2". Two task names are registered, to differentiate between frozen and tuned setting. 

"cola-pair-tuned" compute the acceptability logits on two sentences independently with the classifier 
 trained on cola and choose the more likely one. "cola-pair-frozen" predict the word at the location where two sentences differs with pretrained MaskLM head.

Borrowing classifier from another task was initially introduced only for mnli-diagnostic, therefore the control logic in main.py are written specifically for mnli-diagnostic. This update generalize that mechanism and allow other tasks to borrow classifier from a intrinsically similar task.
To use this feature, set TASK1 {use_classifier = "TASK0"} in default.conf, so that TASK1 will reuse the classifier of TASK0, instead of create its own.

sleepinyourhat commented on config/spring19_seminar/colapair_bertF.conf:
```diff
@@ -0,0 +1,60 @@
+// Direct probe model on cola-pair dataset, the parameters are frozen, we only use MaskLM to do acceptability judgement 
+// F stands for frozen
```
Might as well spell this out in the filename.

HaokunLiu replied:
ok

sleepinyourhat commented on config/spring19_seminar/colapair_bertF.conf:
```diff
@@ -0,0 +1,60 @@
+do_full_eval = 1  // evaluation the model after training
+
+transfer_paradigm = "frozen"
+batch_size = 16
```
FWIW, try to coordinate with others on this. We use 24 elsewhere, and this could cause some issues with comparison.

HaokunLiu replied:
I got batch_size = 16 from final_bert.conf , I couldn't find any bert config that have batch_size = 24.

sleepinyourhat replied:
It looks like that might have just been @PterosDiacos. Fine to stick with 16.

sleepinyourhat commented on config/spring19_seminar/colapair_bertF.conf:
```diff
@@ -0,0 +1,60 @@
+elmo = 0
+sep_embs_for_skip = 1
+tokenizer = "bert-base-uncased"
+bert_model_name = "bert-base-uncased"
```
We use cased in some other experiments. Coordinate with others.

HaokunLiu replied:
Yes. Some part of data(not NPI) is not cased, so I used uncased initially. Now that we have pivoted to focus on NPI, we can safely change to cased.

sleepinyourhat replied:
Sounds good!

sleepinyourhat commented on config/spring19_seminar/colapair_bertF.conf:
```diff
@@ -0,0 +1,60 @@
+                               //                "cat" uses lexical layer + top layer
+bert_fine_tune = 0
+pair_attn = 0 // shouldn''t be needed but JIC
+dropout = 0.1 // following BERT paper
```
If you inherit from one of the BERT config files, will you get this for free? Best to try to add only the changes we really need.

HaokunLiu replied:
Yes, it should be better to import bert-final or make a stem config for spring19_seminar to import from.

sleepinyourhat commented on main.py:
```diff
@@ -313,11 +313,12 @@ def main(cl_arguments):
             # "is_probing_task is global flag specific to a run, not to a task.
-            if task.name == 'mnli-diagnostic':
+            if task.get_split_text('train') == task.get_split_text('val') or \
+                task.get_split_text('train') == task.get_split_text('test'):
```
If this is at all slow, we should just set an attribute of tasks that we can check for.

HaokunLiu replied:
done

sleepinyourhat commented on main.py:
```diff
@@ -368,20 +369,21 @@ def main(cl_arguments):
 
-                tasks = [task]
-                if task.name == 'mnli':
-                    tasks += [t for t in target_tasks if t.name == 'mnli-diagnostic']
```
Make sure to double check that mnli-diagnostic still works the same way.

HaokunLiu replied:
I can do it on Thursday or Friday.
Some part of the code in mnli-diagnostic is unnecessarily unique and long, since we need to test it either way, I wonder can we take this chance to refactor mnli-diagnostic?

sleepinyourhat replied:
Sure, but you'd have to be careful to test it thoroughly.

sleepinyourhat commented on scripts/cola_baseline.sh:
```diff
@@ -12,3 +12,9 @@ python main.py --config_file "config/spring19_seminar/cola_elmo.conf" --override
 
```
Mind moving this to a scripts/spring19-seminar directory?

HaokunLiu replied:
Will do.

sleepinyourhat commented on config/spring19_seminar/cola_gpt.conf:
```diff
@@ -8,6 +8,7 @@ include "cola_elmo.conf"
 do_target_task_training = 0  // only do main training phase, since our pretrain and target task are the same
 allow_reuse_of_pretraining_parameters = 1  // set this to 1 because we pretrain and finetune model on the same task
 batch_size = 2  // keep it small to avoid out-of-memory
+lr = 1e-5  // we need to keep lr small for fine-tuning gpt, especially so because our BS is smaller than that in gpt paper
```
Fine to check this in, but we probably won't be publishing anything with GPT, right?

HaokunLiu replied:
No, we won't

W4ngatang commented on src/bert/utils.py:
```diff
@@ -152,3 +152,13 @@ def get_output_dim(self):
         else:
             return self.model.config.hidden_size
+
+    def hatch_LM_head(self, args):
```
What does `hatch` mean? Could you make the docstring a little clearer as to what this is doing?

HaokunLiu replied:
Well, this name is not very intuitive. I have changed it and added notes

W4ngatang commented on src/models.py:
```diff
@@ -748,6 +753,47 @@ def _single_sentence_forward(self, batch, task, predict):
+            classifier = self._get_classifier(task)
+            index = batch['index'].squeeze(dim=1)
+            logits_full = classifier(sent_embs)[range(bs), index]
+            sent1_key = batch['input1']['bert_wpm_pretokenized'][range(bs), index] - 2
```
this will break whenever you aren't using BERT tokenization?

HaokunLiu replied:
The program would not enter this part of the code, if it isn't using bert. (there won't be suitable task specific module for it, so it will get ValueError("Module not found for %s" % task.name)). This forward function is only for probing BERT with minimal pair, the name of this function did not reflect that, I should probably change it.

W4ngatang replied:
Ah gotcha. Are you all planning on supporting non-BERT versions of this task? Otherwise renaming would be helpful.

HaokunLiu replied:
No. Minimal pairs is specially used as a tool to probe the properties of BERT model. I can't see a scenario where this task is necessary to understand something in other models.

W4ngatang commented on src/tasks/tasks.py:
```diff
@@ -223,6 +223,11 @@ def n_train_examples(self):
 
+    @property
+    def no_train_data(self):
+        return self.get_split_text('train') == self.get_split_text('val') or \
```
isn't this kind of expensive for what could otherwise be a Boolean we pass? I guess it does get kind of tricky if you want to have versions of the same task class that do and do not have training data, but I'm not sure how often that's happening

HaokunLiu replied:
I think that is not a big issue. Time-wise, this command is run only once, won't make any difference. Avoiding setting up another Boolean make it less likely for people to make mistakes when adding new tasks. As for the case of same task have and don't have training data, if the training split is set as convention (like mnli-diagnostic), this function will give us the correct Boolean.

W4ngatang replied:
Hmm okay, fair enough.

W4ngatang commented on src/tasks/tasks.py:
```diff
@@ -543,6 +546,101 @@ def get_metrics(self, reset=False):
+                'mcc',
+                self.tag_list,
+                reset))
+        # collected_metrics.update(
```
might as well delete

W4ngatang commented on src/tasks/tasks.py:
```diff
@@ -543,6 +546,101 @@ def get_metrics(self, reset=False):
+        return instances  # lazy iterator
+    
+    def update_metrics(self, logits, labels, tagmask=None):
+        logits, labels = logits.detach(), labels.detach()
```
do you need to detach explicitly? I thought within scorers we're calling `.item()` or something similar anyways.

W4ngatang commented on src/tasks/tasks.py:
```diff
@@ -543,6 +546,101 @@ def get_metrics(self, reset=False):
                 reset))
         return collected_metrics
 
+@register_task('cola-pair-frozen', rel_path='CoLA')
```
Sorry to be annoying about this, but could you create a new file in `src/tasks/`? I'd like to avoid throwing more tasks, esp. tasks that are somewhat project specific, into this file.

HaokunLiu replied:
It's okay, I agree with you.

W4ngatang commented on src/models.py:
```diff
@@ -422,7 +422,7 @@ def build_task_specific_modules(
      '''
     task_params = model._get_task_params(task.name)
-    if isinstance(task, SingleClassificationTask):
+    if isinstance(task, SingleClassificationTask) or (isinstance(task, CoLAMinimalPairTask) and task.name == 'cola-pair-tuned'):
```
Could you restructure to do the second check within a branch for `isinstance(task, CoLAMinimalPairTask)`? Would like to avoid the check for this branch including very task-specific things (although I know we're guilty of this elsewhere).

HaokunLiu replied:
Sure. (that is so true ; )

W4ngatang commented on src/tasks/tasks.py:
```diff
@@ -800,6 +803,90 @@ def load_data(self, path, max_seq_len):
         log.info("\tFinished loading MNLI data.")
 
 
+@regiscter_task('mnli-diag', rel_path='MNLI/')
```
register

HaokunLiu replied:
done

W4ngatang commented on src/models.py:
```diff
@@ -433,6 +434,16 @@ def build_task_specific_modules(
+            setattr(model, '%s_mdl' % task.name, mask_lm_head)
+        elif task.name == 'cola-pair-tuned':
+            module = build_single_sentence_module(task=task, d_inp=d_sent,
+                                            use_bert=model.use_bert, params=task_params)
```
minor: indentation

HaokunLiu replied:
thanks.

W4ngatang commented on main.py:
```diff
@@ -313,11 +313,11 @@ def main(cl_arguments):
             # to be set to True. For mnli-diagnostic this flag will be False because it is part of GLUE and
             # "is_probing_task is global flag specific to a run, not to a task.
-            if task.name == 'mnli-diagnostic':
+            if task.no_train_data:
```
thanks!

HaokunLiu commented:
No hurry review the PR, I haven’t done testing the new mnli-diagostic yet.

sleepinyourhat commented:
Quick comment (not jumping into the review process): See if you can finish this before Friday. There'll be a bunch of refactors then, so you're likely to hit merge conflicts if this is still unfinished after Friday morning.

HaokunLiu commented:
Finally tested. 
Please let me know if there is anything else we need to change.
@sleepinyourhat @W4ngatang

sleepinyourhat commented on config/spring19_seminar/colapair_bert_frozen.conf:
```diff
@@ -0,0 +1,30 @@
+// Do not change
+
+word_embs_file = ""  // we do not need this in elmo, gpt or bert
+fastText_model_file = ""  // we do not need this in elmo, gpt or bert
```
This is deprecated. Delete?

HaokunLiu replied:
Yes

sleepinyourhat commented on config/spring19_seminar/colapair_bert_frozen.conf:
```diff
@@ -0,0 +1,30 @@
+
+// Model settings
+
+bert_fine_tune = 0
```
Doesn't matter, since we never do training at all.

sleepinyourhat commented on config/spring19_seminar/colapair_bert_frozen.conf:
```diff
@@ -0,0 +1,30 @@
+
+// Do not change
+
+word_embs_file = ""  // we do not need this in elmo, gpt or bert
```
Delete. Even if this is specified, we won't load embeddings here.

W4ngatang commented on src/evaluate.py:
```diff
@@ -111,7 +111,7 @@ def evaluate(model, tasks: Sequence[tasks_module.Task], batch_size: int,
         log.info("Finished evaluating on: %s", task.name)
 
-    all_metrics["micro_avg"] /= n_examples_overall
+    all_metrics["micro_avg"] /= max(n_examples_overall, 1)
```
is this to avoid issues with the diagnostics? If so, leave a comment saying so

HaokunLiu replied:
Yes

W4ngatang commented on src/tasks/tasks.py:
```diff
@@ -457,6 +462,10 @@ def update_metrics(self, logits, labels, tagmask=None):
 
 @register_task('cola-analysis', rel_path='CoLA/')
 class CoLAAnalysisTask(SingleClassificationTask):
```
actually, could you put this into the `minimal_pairs` task file and rename it to something like `tasks/acceptability.py`? I'm trying to avoid growing `tasks.py` any more.

HaokunLiu replied:
Ok

HaokunLiu commented:
We don't need a special config file for minimal pair experiments now, we just need a sh script. So those config are deleted.

HaokunLiu commented:
The script is not complete, I will adjust as needed when running it. 
For now, is it possible to merge this and allow me to upload a final version of the script afterwards?

pep8speaks commented:
Hello @HaokunLiu! Thanks for updating this PR. 

* In the file [`main.py`](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py):

> [Line 20:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L20): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 22:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L22): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 23:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L23): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 24:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L24): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 25:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L25): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 26:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L26): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 27:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L27): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 92:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L92): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 100 characters)
> [Line 104:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L104): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 100 characters)
> [Line 141:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L141): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 100 characters)
> [Line 163:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L163): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 100 characters)
> [Line 320:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L320): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (113 > 100 characters)
> [Line 392:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L392): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 393:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L393): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 100 characters)
> [Line 395:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L395): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (108 > 100 characters)
> [Line 396:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L396): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (119 > 100 characters)
> [Line 397:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L397): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 414:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L414): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 100 characters)
> [Line 416:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L416): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (112 > 100 characters)
> [Line 421:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L421): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (109 > 100 characters)
> [Line 463:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L463): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 100 characters)
> [Line 466:5](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L466): [E125](https://duckduckgo.com/?q=pep8%20E125) continuation line with same indent as next logical line
> [Line 467:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/main.py#L467): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 100 characters)

* In the file [`src/bert/utils.py`](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/bert/utils.py):

> [Line 80:38](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/bert/utils.py#L80): [E252](https://duckduckgo.com/?q=pep8%20E252) missing whitespace around parameter equals
> [Line 80:39](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/bert/utils.py#L80): [E252](https://duckduckgo.com/?q=pep8%20E252) missing whitespace around parameter equals
> [Line 167:26](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/bert/utils.py#L167): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file

* In the file [`src/evaluate.py`](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/evaluate.py):

> [Line 43:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/evaluate.py#L43): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 100 characters)
> [Line 132:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/evaluate.py#L132): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 100 characters)
> [Line 168:22](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/evaluate.py#L168): [E124](https://duckduckgo.com/?q=pep8%20E124) closing bracket does not match visual indentation
> [Line 170:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/evaluate.py#L170): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 214:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/evaluate.py#L214): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 297:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/evaluate.py#L297): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)

* In the file [`src/models.py`](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py):

> [Line 58:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L58): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 100 characters)
> [Line 76:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L76): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (103 > 100 characters)
> [Line 89:26](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L89): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 89:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L89): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 100 characters)
> [Line 90:26](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L90): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 93:37](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L93): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 94:37](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L94): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 95:37](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L95): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 96:37](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L96): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 341:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L341): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (103 > 100 characters)
> [Line 472:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L472): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 100 characters)
> [Line 503:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L503): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (121 > 100 characters)
> [Line 621:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L621): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 100 characters)
> [Line 625:5](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L625): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)
> [Line 626:77](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L626): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 632:48](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L632): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 724:5](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L724): [E125](https://duckduckgo.com/?q=pep8%20E125) continuation line with same indent as next logical line
> [Line 743:81](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L743): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 812:79](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L812): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 814:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L814): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (111 > 100 characters)
> [Line 822:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L822): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 834:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L834): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 843:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L843): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 100 characters)
> [Line 916:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L916): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 100 characters)
> [Line 1008:56](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L1008): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 1054:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L1054): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (108 > 100 characters)
> [Line 1055:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L1055): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 100 characters)
> [Line 1106:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L1106): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (122 > 100 characters)
> [Line 1117:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/models.py#L1117): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (112 > 100 characters)

* In the file [`src/tasks/__init__.py`](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/__init__.py):

> [Line 42:23](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/__init__.py#L42): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 44:46](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/__init__.py#L44): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 45:49](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/__init__.py#L45): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 46:55](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/__init__.py#L46): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 47:58](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/__init__.py#L47): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 48:49](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/__init__.py#L48): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 49:46](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/__init__.py#L49): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 50:49](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/__init__.py#L50): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 51:58](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/__init__.py#L51): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

* In the file [`src/tasks/acceptability.py`](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py):

> [Line 35:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L35): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 50:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L50): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 54:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L54): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (121 > 100 characters)
> [Line 55:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L55): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (116 > 100 characters)
> [Line 56:28](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L56): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 56:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L56): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (124 > 100 characters)
> [Line 76:23](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L76): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 77:81](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L77): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 95:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L95): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 100:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L100): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 109:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L109): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 146:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L146): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 100 characters)
> [Line 148:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L148): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 100 characters)
> [Line 149:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L149): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 100 characters)
> [Line 150:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L150): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (132 > 100 characters)
> [Line 209:33](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/acceptability.py#L209): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file

* In the file [`src/tasks/tasks.py`](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py):

> [Line 85:38](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L85): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 261:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L261): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 342:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L342): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 0
> [Line 367:5](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L367): [E301](https://duckduckgo.com/?q=pep8%20E301) expected 1 blank line, found 0
> [Line 372:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L372): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 399:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L399): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 416:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L416): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 100 characters)
> [Line 430:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L430): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 100 characters)
> [Line 433:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L433): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 100 characters)
> [Line 470:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L470): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 100 characters)
> [Line 480:9](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L480): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 521:9](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L521): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 533:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L533): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (101 > 100 characters)
> [Line 705:9](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L705): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 706:9](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L706): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 715:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L715): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 717:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L717): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 100 characters)
> [Line 747:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L747): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (125 > 100 characters)
> [Line 749:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L749): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (124 > 100 characters)
> [Line 776:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L776): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 100 characters)
> [Line 780:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L780): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (109 > 100 characters)
> [Line 781:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L781): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (113 > 100 characters)
> [Line 782:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L782): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 100 characters)
> [Line 783:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L783): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (103 > 100 characters)
> [Line 792:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L792): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (109 > 100 characters)
> [Line 793:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L793): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 100 characters)
> [Line 794:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L794): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 100 characters)
> [Line 795:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L795): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (109 > 100 characters)
> [Line 796:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L796): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 100 characters)
> [Line 797:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L797): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (109 > 100 characters)
> [Line 819:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L819): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 823:29](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L823): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 824:29](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L824): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 825:29](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L825): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 826:29](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L826): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 851:42](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L851): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 872:5](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L872): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)
> [Line 889:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L889): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 890:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L890): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 892:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L892): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (3)
> [Line 905:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L905): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (108 > 100 characters)
> [Line 907:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L907): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 100 characters)
> [Line 909:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L909): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (108 > 100 characters)
> [Line 1010:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1010): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (107 > 100 characters)
> [Line 1012:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1012): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 100 characters)
> [Line 1014:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1014): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (106 > 100 characters)
> [Line 1165:42](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1165): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 1181:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1181): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 100 characters)
> [Line 1192:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1192): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (117 > 100 characters)
> [Line 1194:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1194): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (112 > 100 characters)
> [Line 1196:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1196): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (112 > 100 characters)
> [Line 1454:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1454): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 1473:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1473): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (108 > 100 characters)
> [Line 1474:15](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1474): [E222](https://duckduckgo.com/?q=pep8%20E222) multiple spaces after operator
> [Line 1474:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1474): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (123 > 100 characters)
> [Line 1475:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1475): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 100 characters)
> [Line 1479:5](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1479): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)
> [Line 1480:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1480): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (116 > 100 characters)
> [Line 1481:27](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1481): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 1481:72](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1481): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 1481:74](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1481): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 1481:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1481): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (149 > 100 characters)
> [Line 1482:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1482): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 100 characters)
> [Line 1483:74](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1483): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 1483:76](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1483): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 1483:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1483): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (150 > 100 characters)
> [Line 1484:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1484): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (115 > 100 characters)
> [Line 1485:73](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1485): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 1485:75](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1485): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 1485:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1485): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (134 > 100 characters)
> [Line 1491:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1491): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 100 characters)
> [Line 1499:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1499): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (129 > 100 characters)
> [Line 1500:50](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1500): [E712](https://duckduckgo.com/?q=pep8%20E712) comparison to True should be 'if cond is True:' or 'if cond:'
> [Line 1512:1](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1512): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 1529:26](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1529): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 1536:9](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1536): [E306](https://duckduckgo.com/?q=pep8%20E306) expected 1 blank line before a nested definition, found 0
> [Line 1540:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1540): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 100 characters)
> [Line 1541:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/tasks/tasks.py#L1541): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 100 characters)

* In the file [`src/utils/data_loaders.py`](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/utils/data_loaders.py):

> [Line 29:23](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/utils/data_loaders.py#L29): [E261](https://duckduckgo.com/?q=pep8%20E261) at least two spaces before inline comment
> [Line 99:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/utils/data_loaders.py#L99): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 100 characters)
> [Line 100:101](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/utils/data_loaders.py#L100): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (122 > 100 characters)
> [Line 123:14](https://github.com/nyu-mll/jiant/blob/6f3ce4b2554e4e1f5d13e57c92a0b75750cc21e1/src/utils/data_loaders.py#L123): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

You can automatically repair most issues by commenting `@pep8speaks pep8ify` on GitHub or by locally running: `python -m autopep8 --max-line-length 100  --in-place --aggressive  --aggressive  ./*.py ./*/*.py ./*/*/*.py`

HaokunLiu commented:
Some updates:
1. Since mnli diagnostic task have some changes in SuperGLUE, it is safer to keep it as is. So I removed my version of the mnli diagnostic task in the code.
2. I moved NPI task to acceptablity.py

sleepinyourhat commented:
Thanks! Make sure that tests all pass and let me know when it’s ready for review.

HaokunLiu commented on src/utils/data_loaders.py:
```diff
@@ -26,12 +26,12 @@ def load_span_data(tokenizer_name, file_name, label_fn=None, has_labels=True):
     it becomes [0, 3].
     The task file should of be of the following form: 
-        text: str, 
+        text: str
```
Strictly speaking, this is not something I should touch, since it's someone else's  function. But the format checker complained, and the following comma is inconsistent with everywhere else, it is not too terrible to make it cleaner, right?

sleepinyourhat replied:
Fine by me!

sleepinyourhat commented on src/bert/utils.py:
```diff
@@ -149,3 +149,16 @@ def get_output_dim(self):
+
+    def transplant_LM_head(self, args):
+        """
+        download the masked LM classifier from huggingface bert
```
Clean this up to match other fns here: Use a full sentence with capitalization. Otherwise looks good!

sleepinyourhat commented on src/tasks/acceptability.py:
```diff
@@ -0,0 +1,433 @@
+
+@register_task("npi_pair_frozen", rel_path="NPI")
+class NPIMinimalPairFrozenTask(Task):
+    """ Task class for frozen minimal pair (cloze) acceptability judgement """
```
How about 'NPIMinimalPairMLMTask' - MLM vs. classifier seems more important than frozen vs. tuned.

HaokunLiu replied:
Yes, that's good, I'll change this. This naming is pretty bad, especially so when considering that "frozen" is also used as a keyword for transfer paradigm.

sleepinyourhat commented on src/tasks/acceptability.py:
```diff
@@ -0,0 +1,433 @@
+"""Define the tasks and code for loading their data.
```
This comment seems wrong. Did you just copy it from somewhere else? It need to explain what this file is for, and when someone should use it.
 
Also, if CoLA isn't here, 'acceptability' isn't a good name for the file. 'acceptability_probing.py'?

HaokunLiu replied:
(1) Exactly.
(2) Sure.

sleepinyourhat commented on src/tasks/acceptability.py:
```diff
@@ -0,0 +1,433 @@
+            count=len(self.tag_list), scorer_type=CategoricalAccuracy
+        )
+
+        log.info("\tFinished loading CoLA sperate domain.")
```
I don't know what this means. Clarify.

sleepinyourhat commented on src/tasks/acceptability.py:
```diff
@@ -0,0 +1,433 @@
+
+    def process_split(self, split, indexers):
+        def _make_instance(input1, input2, labels, tagids):
+            """ from multiple types in one column create multiple fields """
```
Use full sentences with punctuation.

HaokunLiu replied:
This was initially from process_split in tasks.py, since it's no longer there, I guess it's fine to remove this comment here as well?

sleepinyourhat commented on src/tasks/acceptability.py:
```diff
@@ -0,0 +1,433 @@
+@register_task("hd_cola_npi_sup", rel_path="NPI/combs/minus_superlative")
+class CoLANPITask(SingleClassificationTask):
+    """Class for NPI-related task; same with Warstdadt acceptability task but outputs labels for
+       test-set
```
Typo in Alex's name. Use full sentence.

HaokunLiu replied:
Ah, that's embarrassing, I should probably have noticed this when moving this code from tasks.py.

sleepinyourhat commented on src/tasks/acceptability.py:
```diff
@@ -0,0 +1,433 @@
+
+        self.val_metric = "%s_mcc" % self.name
+        self.val_metric_decreases = False
+        # self.scorer1 = Average()
```
Cut.

sleepinyourhat commented on src/tasks/tasks.py:
```diff
@@ -232,6 +232,12 @@ def n_train_examples(self):
+    def no_train_data(self):
+        return self.get_split_text("train") == self.get_split_text("val") or self.get_split_text(
+            "train"
+        ) == self.get_split_text("test")
```
This is too dangerous: It's a common and easy sanity check to do an experiment where they load the same file for training and validation and see if the model can get to 100% on that data. This would break that setup. 

Find another way to do this—maybe set it to False in the general task class, and override it explicitly when you need it.

HaokunLiu replied:
Yes, I agree.

HaokunLiu commented:
This PR is now in an awkward situation. I changed the code after last review. But somehow the PR is still in "change-required", so it didn't received any further review for a long time.  This PR is a conglomeration of multiple things, that makes the commit-review-merge cycle on it quite slow. Now, jiant has gone through some major update. Some parts of this PR is still relevant, the other are no longer needed. It is not very convenient to merge things together.

How about, alternatively, I make several smaller independent PRs when working on Alex Warstadt's summer project, absorbing still necessary parts in this PR into those PRs. I think, this is the easiest way to make the features in this PR catch up with main jiant, and avoid any more prolonged PR processes.

sleepinyourhat commented:
Yes, that sounds good. Thanks!

- Make sure the future PRs take into account the comments from here.
- Only submit a PR in the future if you're willing to finish it within a couple of weeks.
